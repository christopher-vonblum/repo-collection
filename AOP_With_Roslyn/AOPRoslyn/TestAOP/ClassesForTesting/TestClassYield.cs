using System;
using System.Collections.Generic;
using System.Text;

namespace TestAOP.ClassesForTesting
{
    class TestClassYield
    {
       
        public string FirstName { get; set; }
        public string LastName { get; set; }

        public string Name(string separator = "")
        {
            return FirstName + separator + LastName;
        }


        public IEnumerable<string> Test()
        {
            yield return FirstName;
            yield return LastName;
        }
        
    }
}
