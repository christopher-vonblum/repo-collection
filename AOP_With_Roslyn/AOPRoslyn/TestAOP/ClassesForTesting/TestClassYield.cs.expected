using System;
using System.Collections.Generic;
using System.Text;

namespace TestAOP.ClassesForTesting
{
    class TestClassYield
    {
       
        public string FirstName { get; set; }
        public string LastName { get; set; }

        public string Name(string separator = "")
{System.Console.WriteLine("start TestClassYield_Name_12");System.Console.WriteLine("separator="+(separator??"").ToString());
#line 15
return FirstName + separator + LastName;
System.Console.WriteLine("end TestClassYield_Name_12");}

        public IEnumerable<string> Test()
        {
            yield return FirstName;
            yield return LastName;
        }
        
    }
}
